tests/e2e/specs/shopper/filter-products-by-attribute.test.ts

```diff
@@ -0,0 +1,205 @@
+import {
+	canvas,
+	createNewPost,
+	deleteAllTemplates,
+	insertBlock,
+	switchUserToAdmin,
+	publishPost,
+} from '@wordpress/e2e-test-utils';
+import { Frame } from 'puppeteer';
+import { selectBlockByName } from '@woocommerce/blocks-test-utils';
+
+import {
+	BASE_URL,
+	goToTemplateEditor,
+	openBlockEditorSettings,
+	saveTemplate,
+	useTheme,
+} from '../../utils';
+
+const block = {
+	name: 'Filter Products by Attribute',
+	slug: 'woocommerce/attribute-filter',
+	class: '.wc-block-attribute-filter',
+	selectors: {
+		editor: {
+			firstAttributeInTheList:
+				'.woocommerce-search-list__list > li > label > input.woocommerce-search-list__item-input',
+			filterButtonToggle: "//label[text()='Filter button']",
+			doneButton: '.wc-block-attribute-filter__selection > button',
+		},
+		frontend: {
+			firstAttributeInTheList:
+				'.wc-block-attribute-filter-list > li:not([class^="is-loading"])',
+			productsList: '.wc-block-grid__products > li',
+			classicProductsList: '.products.columns-3 > li',
+			filter: "input[id='128gb']",
+			submitButton: '.wc-block-components-filter-submit-button',
+		},
+	},
+	urlSearchParamWhenFilterIsApplied:
+		'?filter_capacity=128gb&query_type_capacity=or',
+};
+
+const waitForAllProductsBlockLoaded = () =>
+	page.waitForSelector( selectors.frontend.productsList + '.is-loading', {
+		hidden: true,
+	} );
+
+const goToShopPage = () =>
+	page.goto( BASE_URL + '/shop', {
+		waitUntil: 'networkidle0',
+	} );
+
+const { selectors } = block;
+
+describe( `${ block.name } Block`, () => {
+	describe( 'with All Products Block', () => {
+		beforeAll( async () => {
+			await switchUserToAdmin();
+			await createNewPost( {
+				postType: 'post',
+				title: block.name,
+			} );
+
+			await insertBlock( block.name );
+			await page.click( selectors.editor.firstAttributeInTheList );
+			await page.click( selectors.editor.doneButton );
+			await insertBlock( 'All Products' );
+			await publishPost();
+
+			const link = await page.evaluate( () =>
+				wp.data.select( 'core/editor' ).getPermalink()
+			);
+
+			await page.goto( link );
+		} );
+
+		it( 'should render', async () => {
+			await waitForAllProductsBlockLoaded();
+			const products = await page.$$( selectors.frontend.productsList );
+
+			expect( products ).toHaveLength( 4 );
+		} );
+
+		it( 'should show only products that match the filter', async () => {
+			const isRefreshed = jest.fn( () => void 0 );
+			page.on( 'load', isRefreshed );
+			await page.click( selectors.frontend.filter );
+			await waitForAllProductsBlockLoaded();
+			const products = await page.$$( selectors.frontend.productsList );
+
+			expect( isRefreshed ).not.toBeCalled();
+			expect( products ).toHaveLength( 1 );
```

Comment:
Mentioned this further but first :) if we are testing for a filter match we should check it's the product we expect to ensure we didn't get the right count but the wrong content.